platform-api/che-core-api-ssh/src/main/java/org/eclipse/che/api/ssh/server/SshService.java

```diff
@@ -0,0 +1,274 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.ssh.server;
+
+import com.wordnik.swagger.annotations.Api;
+import com.wordnik.swagger.annotations.ApiOperation;
+import com.wordnik.swagger.annotations.ApiParam;
+import com.wordnik.swagger.annotations.ApiResponse;
+import com.wordnik.swagger.annotations.ApiResponses;
+
+import org.apache.commons.fileupload.FileItem;
+import org.eclipse.che.api.core.BadRequestException;
+import org.eclipse.che.api.core.ConflictException;
+import org.eclipse.che.api.core.NotFoundException;
+import org.eclipse.che.api.core.ServerException;
+import org.eclipse.che.api.core.rest.Service;
+import org.eclipse.che.api.core.rest.annotations.GenerateLink;
+import org.eclipse.che.api.core.rest.shared.dto.Link;
+import org.eclipse.che.api.core.util.LinksHelper;
+import org.eclipse.che.api.ssh.server.model.impl.SshPairImpl;
+import org.eclipse.che.api.ssh.shared.Constants;
+import org.eclipse.che.api.ssh.shared.dto.GenerateSshPairRequest;
+import org.eclipse.che.api.ssh.shared.dto.SshPairDTO;
+import org.eclipse.che.api.ssh.shared.model.SshPair;
+import org.eclipse.che.commons.env.EnvironmentContext;
+
+import javax.annotation.security.RolesAllowed;
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static org.eclipse.che.api.ssh.shared.Constants.LINK_REL_GET_PAIR;
+import static org.eclipse.che.api.ssh.shared.Constants.LINK_REL_REMOVE_PAIR;
+import static org.eclipse.che.dto.server.DtoFactory.newDto;
+
+/**
+ * Defines Ssh Rest API.
+ *
+ * @author Sergii Leschenko
+ */
+@Api(value = "/ssh", description = "Ssh REST API")
+@Path("/ssh")
+public class SshService extends Service {
+    private final SshManager sshManager;
+
+    @Inject
+    public SshService(SshManager sshManager) {
+        this.sshManager = sshManager;
+    }
+
+    @POST
+    @Path("generate")
+    @Consumes(APPLICATION_JSON)
+    @Produces(APPLICATION_JSON)
+    @RolesAllowed("user")
+    @GenerateLink(rel = Constants.LINK_REL_GENERATE_PAIR)
+    @ApiOperation(value = "Generate and stores ssh pair based on the request",
+                  notes = "This operation can be performed only by authorized user," +
+                          "this user will be the owner of the created ssh pair",
+                  response = SshPairDTO.class)
```

Comment:
It is not required to define response implicitly when the method return type is not `Response`